buildtest/builders/base.py

```diff
@@ -930,6 +930,26 @@ def _get_variables(self, variables):
 
         return lines
 
+    def _extract_content(self, regex):
+        """Extract content based on the stream and linenum properties
+        from the regex field and return it as a string.
+
+        Args:
+            regex (dict): regex section from the metrics field
+        """
+
+        stream = regex.get("stream")
+        content = self._output if stream == "stdout" else self._error
+
+        linenum = regex.get("linenum")
+        if linenum is not None and content:
+            lines = content.split("\n")
+            try:
+                content = lines[:-1][linenum] if lines[-1] == "" else lines[linenum]
+            except Exception as e:
+                raise ValueError("Failed to index linenum from the content") from e
+        return content
```

Comment:
i dont really follow this logic for `content = lines[-1][linenum] ...`

I think what we need is a `try/except` block to fetch index into `lines` and retrieve, if there is an IndexError then return an empty string since metrics would be undefined.

we should change `Exception` to `IndexError` since that is what would get raised when you access an index.

Also we should not raise `ValueError` since this will prevent the test from running, rather we should set the value to empty string `""`.

That being said i think the code should be something like

```python
try:
  content = lines[linenum]
except IndexError as e
  content = ''
  logger.error(e)
return content
```

Reply:
@Xiangs18 Please try my suggestion and try running the examples to see if it works. The `logger` variable should be accessible, it might be `self.logger` also so try running the invalid example and then check the log output for error message.